Vignette Introduction (#150)
terraref/tutorials: vignettes/00-introduction.Rmd

# (PART\*) Section 1: Vignettes {-}

# Vignettes Introduction

> These vignettes provide a **quick start** tutorial - they quickly show you how to access and perform ‘hello world’ type introductions to different types of data in the TERRA Reference datasets. These vignettes are designed so that they don’t require any user accounts or passwords to access to the data so that they can be run as easily as possible.

Reviewer: ```suggestion
These vignettes provide a **quick start** tutorial - they quickly show you how to access different types of data in the TERRA Reference datasets and perform ‘hello world’ tasks with them. 
```

> The vignettes currently include:

Reviewer: ```suggestion
The vignettes currently include examples of:
```

* trait data
* weather data
* image data

> * how to use combine trait, weather, and image data in a single analysis

Reviewer: ```suggestion
* combining trait, weather, and image data in a single analysis
```

> In order to make these vignettes as easy to use as possible, we only use sample of the full dataset is available without an account; specifically we have made data from thirty two plots in each of two seasons available. Although we want to make these data broadly available, we want to keep track of who is using them and collect emails so that we can contact you.

Reviewer: ```suggestion
In order to make these vignettes as easy to use as possible, we are using a subset of the full dataset that is is available without an account. Specifically, we have made data from thirty two plots in each of the two seasons available. 

Although we want to make these data broadly available, the remainder of the dataset requires an account to use so we can keep track of who and how many people are using it, and can contact users with their provided email addresses. Information for getting a beta user account is available in [section 1.1.2](https://terraref.github.io/tutorials/index.html). 
```

> In order to use the vignettes, you will need to have R installed along with the extra packages that are used. Except where otherwise indicated, the packages are available on CRAN and can be installed using the syntax `install.packages('packagename')`. 

Reviewer: They'll also need Python for the third and fourth vignettes?

Author: ```suggestion
In order to use the vignettes, you will need to have R installed along with the extra packages that are used. 
All of the packages except 'trait' are available on CRAN and can be installed using the syntax `install.packages('packagename')`:

```r
install.packages('tidyverse', 'jsonlite', 'knitr', 'lubridate', 'raster', 'sf')
```

Although 'traits' is also on CRAN, you need to download the TERRA REF version from GitHub to handle larger datasets: 

```r
devtools::install_github('terraref/traits')
```
```

The vignettes currently include:
* trait data
* weather data
* image data
* how to use combine trait, weather, and image data in a single analysis

In order to make these vignettes as easy to use as possible, we only use sample of the full dataset is available without an account; specifically we have made data from thirty two plots in each of two seasons available. Although we want to make these data broadly available, we want to keep track of who is using them and collect emails so that we can contact you.

## Prerequisites

### R

In order to use the vignettes, you will need to have R installed along with the extra packages that are used. 
All of the packages except 'trait' are available on CRAN and can be installed using the `install.packages` funciton:

```r
install.packages('tidyverse', 'jsonlite', 'knitr', 'lubridate', 'raster', 'sf')
```

Although 'traits' is also on CRAN, you need to download the TERRA REF version from GitHub to handle larger datasets: 

```r
install.packages('devtools')
devtools::install_github('terraref/traits')
```
### Python

> ... need instructions on how to install terrautils! ... 

Author: @Chris-Schnaufer or @KristinaRiemer could one of you put instructions for installing terrautils here?

Reviewer: There are instructions in the images vignette (https://terraref.github.io/tutorials/retrieve-source-rgb-image-files.html), do we want them in both places? I think it makes sense for this intro page to not have installation instructions, and to leave specific install stuff in the relevant vignette.

Reviewer: And I'm only thinking that because it'll be easier to keep installation instructions up to date in each of the vignettes. But I don't have super strong feelings about it.

Author: I think having it in the introduction would be better; otherwise it will be necessary to make sure that installation instructions are in every vignette. 

 ```suggestion
pip install terrautils
```

# (PART\*) Section 1: Vignettes {-}

# Vignettes Introduction

These vignettes provide a **quick start** tutorial - they quickly show you how to access different types of data in the TERRA Reference datasets and perform ‘hello world’ tasks with them. 

The vignettes currently include examples of:
* trait data
* weather data
* image data
* combining trait, weather, and image data in a single analysis

In order to make these vignettes as easy to use as possible, we are using a subset of the full dataset that is is available without an account or password. Specifically, we have made data from thirty two plots in each of the two seasons available. 

> Although we want to make these data broadly available, the remainder of the dataset requires an account to use so we can keep track of who and how many people are using it, and can contact users with their provided email addresses. Information for getting a beta user account is available in [section 1.1.2](https://terraref.github.io/tutorials/index.html). 

Author: ```suggestion
Although we want to make these data broadly available, the remainder of the dataset requires an account to use so we can keep track of who and how many people are using it, and can contact users with their provided email addresses. 
Information for getting a beta user account is available in the section on [User Accounts](#user-accounts).
```

@KristinaRiemer could you please check that this is the correct way to use the internal referencing system described in https://bookdown.org/yihui/bookdown/cross-references.html

Reviewer: @dlebauer that section reference formatting looks correct.

## Prerequisites

### R

In order to use the vignettes, you will need to have R installed along with the extra packages that are used. 
All of the packages except 'trait' are available on CRAN and can be installed using the `install.packages` funciton:

> ```r

Reviewer: All of the little code snippets should have this format, right? 
`` ` `` `` ` `` `` ` ``{r install_r_pks}

Reviewer: (Sorry for the sloppy escaping of the back ticks)

Author: I wasn't sure - in this case it shouldn't be run when the code is compiled, so if we did it should be `{r install_r_pks, eval = FALSE}`

install.packages('tidyverse', 'jsonlite', 'knitr', 'lubridate', 'raster', 'sf')
```

Although 'traits' is also on CRAN, you need to download the TERRA REF version from GitHub to handle larger datasets: 

```r
install.packages('devtools')
devtools::install_github('terraref/traits')
```
### Python

> ... need instructions on how to install terrautils! ... 

Reviewer: ```suggestion

```{sh eval=FALSE}
pip install terrautils
```
```
